geometry/geometry_config.py

```diff
@@ -283,70 +313,29 @@
     c.muShield.LE = 7 * u.m     # - 0.5 m air - Goliath: 4.5 m - 0.5 m air - nu-tau mu-det: 3 m - 0.5 m air. finally 10m asked by Giovanni
     c.muShield.dZ0 = 1 * u.m
 
-    c.muShieldStepGeo = muShieldStepGeo
-    c.muShieldWithCobaltMagnet = muShieldWithCobaltMagnet
 
     # zGap to compensate automatic shortening of magnets
     zGap = 0.05 * u.m  # halflengh of gap
-    if muShieldDesign == 7:
-        c.muShield.Field = 1.8  # Tesla
-        c.muShield.dZ1 = 0.7 * u.m + zGap
-        c.muShield.dZ2 = 1.7 * u.m + zGap
-        c.muShield.dZ3 = 2.0*u.m + zGap
-        c.muShield.dZ4 = 2.0*u.m + zGap
-        c.muShield.dZ5 = 2.75*u.m + zGap
-        c.muShield.dZ6 = 2.4*u.m + zGap
-        c.muShield.dZ7 = 3.0*u.m + zGap
-        c.muShield.dZ8 = 2.35*u.m + zGap
-        c.muShield.dXgap = 0.*u.m
-    elif muShieldDesign == 9:
-        c.muShield.dZ1 = 0.35 * u.m + zGap
-        c.muShield.dZ2 = 2.26 * u.m + zGap
-        c.muShield.dZ3 = 2.08 * u.m + zGap
-        c.muShield.dZ4 = 2.07 * u.m + zGap
-        c.muShield.dZ5 = 2.81 * u.m + zGap
-        c.muShield.dZ6 = 2.48 * u.m + zGap
-        c.muShield.dZ7 = 3.05 * u.m + zGap
-        c.muShield.dZ8 = 2.42 * u.m + zGap
-        c.muShield.dXgap = 0. * u.m
-        c.muShield.half_X_max = 179 * u.cm
-        c.muShield.half_Y_max = 317 * u.cm
-    elif muShieldDesign == 8:
-        assert shieldName
-        params = shield_db[shieldName]
-        c.muShield.params = params
-        c.muShield.dZ1 = 0.35*u.m + zGap
-        c.muShield.dZ2 = 2.26*u.m + zGap
-        c.muShield.dZ3 = params[2]
-        c.muShield.dZ4 = params[3]
-        c.muShield.dZ5 = params[4]
-        c.muShield.dZ6 = params[5]
-        c.muShield.dZ7 = params[6]
-        c.muShield.dZ8 = params[7]
-        c.muShield.dXgap = 0.*u.m
-
-        offset = 7
-        c.muShield.half_X_max = 0
-        c.muShield.half_Y_max = 0
-        for index in range(2, 8):
-            f_l = params[offset + index * 6 + 1]
-            f_r = params[offset + index * 6 + 2]
-            h_l = params[offset + index * 6 + 3]
-            h_r = params[offset + index * 6 + 4]
-            g_l = params[offset + index * 6 + 5]
-            g_r = params[offset + index * 6 + 6]
-            c.muShield.half_X_max = max(c.muShield.half_X_max, 2 * f_l + g_l, 2 * f_r + g_r)
-            c.muShield.half_Y_max = max(c.muShield.half_Y_max, h_l + f_l, h_r + f_r)
-        c.muShield.half_X_max += 15 * u.cm
-        c.muShield.half_Y_max += 15 * u.cm
-    else:
-        raise NotImplementedError(f"muShieldDesign {muShieldDesign} not supported")
+    
+    
+    assert shieldName
```

Comment:
This should be before the first use of shieldName.

It might be good to raise a proper error message instead of the assert.